examples/rpc-client/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsValue;
+
+use wasm_bindgen_futures::JsFuture;
+use worker::*;
+
+#[wasm_bindgen]
```

Comment:
Ideally all of this could be code-gen from capnp or some other specification

Reply:
A general specification makes a lot of sense because of compatibility with implementations in JS.

But an implementation based on the tool rustdocs uses, like pavex does, might be more friendlier as it has the types from the rust side to generate a client crate.

On another perspective, I'm not sure if using the WASM Component Model, we couldn't target the same interface, allowing you to link the client, with the proxy or the real component, avoiding serialisation, a complete bypass.

In theory I think so, in practice, I believe so, if I recall correctly, it might be just a question of linking to different implementations at load time.

But the same tool can be used to generate capnp or the WIT files.

Its even possible to think of when you deploy a Service / DO, its public interface is processed and at runtime its linked directly to it or to a proxy generated on the fly.

So yes, in theory, the same trickery could allow direct C++ -> JS / Rust / Python, Rust -> Rust or Rust -> Proxy -> Rust.

Although I believe it should be feasible right now, much of the path seems the same, even if it isn't just yet for some reason.

In a recent talk at wasmIO they mentioned that C++ support was close by.
But regarding the Component Model in V8, I payed attention, but I didn't capture any mention, and they talked a lot about wasm at Google, V8, wasm gc...

Reply:
Which tool is being used by rustdocs/pavex? They are both large projects. 

I have considered defining the types in Rust, but that makes it very difficult to support multiple languages. The idea for using WIT is very interesting. I will give it some thought.

Reply:
He uses rustdoc-json and rustdoc-types, the same used by cargo-semver-checks and cargo-check-external-types.

He explains how it does it at minute 41m https://www.youtube.com/watch?v=cMea6IMRk2s

He also uses a tool to generate the SDK crate, cargo px, he explains it at min 35.

The ideal would be an equivalent for C++, but if there isn't one, and that's the blocker, the types in rust can work as a proxy. If necessary you can use the comments to do some kind of annotation, so that when you process them you have all the info you require.

For the runtime proxy, there is always the option to generate it from the wit files, I'm just not sure if generating in rust would limit non rust languages.